bears/general/IndentationBear.py

```diff
@@ -60,14 +65,13 @@ def run(self,
             return
 
         insert = ' '*tab_width if use_spaces else '\t'
+        no_indent_file = [line.lstrip() if line.lstrip() is not "" else "\n"
+                          for line_nr, line in enumerate(file)]
+
         new_file = []
-        for line_nr, line in enumerate(file):
-            # Leave out empty lines
-            if line.lstrip() != '':
-                new_file.append(insert*indent_levels[line_nr] +
-                                line.lstrip())
-            else:
-                new_file.append('\n')
+        for line_nr, line in enumerate(no_indent_file):
+            new_file.append(insert*indent_levels[line_nr] +
+                            line if line is not "\n" else "\n")
```

Comment:
The line break confuses me. It looks like you add the indents and then (in the next line) either the line or a new line.

so at least please put the first `line` into the line above and maybe a line break in front of the `else`?
just something that underlines your intent